dotCMS/src/integration-test/java/com/dotcms/security/secret/SecretsStoreKeyStoreImplTest.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    @Test
+    public void Test_404_Cache() {
+
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+
+        final String key = UUIDGenerator.generateUuid();
+        final String value = UUIDGenerator.generateUuid();
+
+        Optional<char[]> noValue = secretsStore.getValue(key);
+
+        assertFalse(noValue.isPresent());
+
+        noValue = secretsStore.getValue(key);
+
+        assertFalse(noValue.isPresent());
+
+        final char[] CACHE_404 = (char[]) CacheLocator.getCacheAdministrator()
+                .getNoThrow(key, SecretsStoreKeyStoreImpl.SECRETS_CACHE_GROUP);
+        assertNotNull(CACHE_404);
+        assertEquals(CACHE_404, SecretsStoreKeyStoreImpl.CACHE_404);
+
+        secretsStore.saveValue(key, value.toCharArray());
+        final Optional<char[]> optionalChars = secretsStore.getValue(key);
+        assertTrue(optionalChars.isPresent());
+        final String returnValue = new String(optionalChars.get());
+
+        assertEquals(value, returnValue);
+    }
+
+
+    @Test
+    public void Test_Deleting_A_Value() {
+
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+
+        final String key = UUIDGenerator.generateUuid();
+        final String value = UUIDGenerator.generateUuid();
+        secretsStore.saveValue(key, value.toCharArray());
+
+        assert (secretsStore.getValue(key).isPresent());
+        secretsStore.deleteValue(key);
+        assert (!secretsStore.getValue(key).isPresent());
+
+    }
+
+
+    @Test
+    public void Test_Empty_Value() {
+        final String uuid = UUIDGenerator.generateUuid();
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+        final Optional<char[]> val = secretsStore.getValue(uuid);
+        assert (!val.isPresent());
+    }
+
+    /**
+     * tests to make sure that the listKeys method returns newly saved keys in their list
+     */
+    @Test
+    public void Test_Value_List() {
+
+        final SecretsStore secretsStore = SecretsStore.INSTANCE.get();
+
+        final String key = UUIDGenerator.generateUuid();
+        final String value = UUIDGenerator.generateUuid();
+
+        secretsStore.saveValue(key, value.toCharArray());
+
+        final String key2 = UUIDGenerator.generateUuid();
+        final String value2 = UUIDGenerator.generateUuid();
+
+        secretsStore.saveValue(key2, value2.toCharArray());
+
+        final String key3 = RandomStringUtils.randomAlphanumeric(1024);
+        final String value3 = RandomStringUtils.randomAlphanumeric(1024);
+
+        secretsStore.saveValue(key3, value3.toCharArray());
+
+        final Collection<String> keys = secretsStore.listKeys();
+        assert (keys.size() > 2);
+        assert (keys.contains(key.toLowerCase()));
+        assert (keys.contains(key2.toLowerCase()));
+        assert (keys.contains(key3.toLowerCase()));
+    }
+
+
+    @Test
+    public void Test_Encryption() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4623512)